S/SymEngine/build_tarballs.jl

```diff
@@ -0,0 +1,58 @@
+# Note that this script can accept some limited command-line arguments, run
+# `julia build_tarballs.jl --help` to see a usage message.
+using BinaryBuilder
+
+# Collection of sources required to build SymEngine
+name = "SymEngine"
+version = v"0.5.0"
+sources = [
+    "https://github.com/symengine/symengine/releases/download/v$(version)/symengine-$(version).tar.gz" =>
+    "5d02002f00d16a0928d1056e6ecb8f34fd59f3bfd8ed0009a55700334dbae29b",
+]
+
+# Bash recipe for building across all platforms
+
+script = raw"""
+cd $WORKSPACE/srcdir/symengine-*
+mkdir build
+cd build
+cmake -DCMAKE_INSTALL_PREFIX=${prefix} \
+      -DCMAKE_TOOLCHAIN_FILE=${CMAKE_TARGET_TOOLCHAIN} \
+      -DBUILD_TESTS=no \
+      -DBUILD_BENCHMARKS=no \
+      -DBUILD_SHARED_LIBS=yes \
+      -DWITH_MPC=yes \
+      -DWITH_SYMENGINE_THREAD_SAFE=yes ..
+make -j${nproc}
+make install
+
+# On Windows, make sure non-versioned filename exists...
+if [[ ${target} == *mingw* ]]; then
+    cp -v ${prefix}/bin/libsymengine-*.dll ${prefix}/bin/libsymengine.dll
```

Comment:
Note that if you add multiple product names like
```
LibraryProduct(["libsymengine", "libsymengine-123"], :libsymengine)
```
(this is an example, I don't know what's the version number here) you don't have to manually move the files